### ISSUE TITLE
Redo operation breaks after a few operations.

### ISSUE THREAD

**Issue by gwk** [Microsoft/vscode#65969]
- VSCode Version:
Version 1.31.0-insider (1.31.0-insider)
7805023eaa46db45d3c24a4c38c0d07c087cf1b3 
2019-01-03T06:06:24.931Z

- OS Version:
macOS 10.14.2 (18C54)

Steps to Reproduce:
1. Launch with extensions disabled. `code --disable-extensions`.
2. Create a new, blank file.
3. Type "abc".
4. Press `cmd+z` to undo. It works.
5. Press `shift+cmd+z` to redo. It works the first time.
6. Try a few more combinations of typing, then undo and redo.
7. After a few steps, redo will fail to do anything. I cannot pin down perfectly reproducible failure, but after a few operations redo is permanently broken until application restart.

The problem seems to bleed across file/text views. Once redo is broken, it is broken application wide. The brokenness also persists across unsaved files that are restored on application launch.

Note: this appears distinct from #65452. I am not sure what a "webview" is as referenced in that issue, but the problem I see is happening in the main text editing view, and is limited to the 'redo' operation.

Does this issue occur when all extensions are disabled?: Yes

**Comment by alexandrudima** [comment]
@gwk I have tried for a couple of minutes, but I cannot reproduce. Redo and Undo work consistently well for me. Do you have any user settings that might impact things? 

There are some settings which lead to edits being created on save, e.g. `files.trimTrailingWhitespace` and others, but even those would just mess with the "redo" stack and never with the "undo" stack.

Please let me know if you find any clearly reproducible steps, I would be very interested to figure this one out.

**Comment by gwk** [comment]
Sorry for the delay, and thanks for taking interest. I have since updated my insiders installation and the bug has disappeared. I have a rather extensive keybindings customization, but I also moved it aside and verified that the problem persisted with no keybindings and extensions off.

Your theory about “edits created on save” and similar bindings does make sense, because indeed undo worked fine at it was only redo that was broken. I imagine this is not wort pursuing further, but if I see it again I will reopen. Thanks again!
